source/Engine.cpp

```diff
@@ -1238,6 +1238,10 @@ void Engine::EnterSystem()
 	// Remove expired bribes, clearance, and grace periods from past fines.
 	GameData::SetDate(today);
 	GameData::StepEconomy();
+
+	// Refresh random systems that could be linked to this one.
+	const_cast<System *>(system)->UpdateRandomLinks(previousSystem);
+
```

Comment:
Note that doing this here and just for one system means that the map can be wrong for _every other system._

Reply:
I know.
It's done this way for optimization's sake, to not render all the map when we dont need to. That said I could just call the UpdateSystems() method if we don't care about that that much?

Reply:
Well I made it redefine the map for all systems at each jump (bye optimization xD)
Only way to make it works without a public updateRandomLinks